src/Cms/Site.php

```diff
@@ -640,9 +640,15 @@ public function visit($page, string|null $languageCode = null)
 			throw new InvalidArgumentException('Invalid page object');
 		}
 
+		// trigger before hook
+		$this->kirby()->trigger('page.visit:before', compact('page', 'languageCode'));
+
 		// set the current active page
 		$this->setPage($page);
 
+		// trigger after hook
+		$this->kirby()->trigger('page.visit:after', compact('page', 'languageCode'));
+
```

Comment:
Maybe we should add the `page.visit` hook at the beginning and end of the `page::render()` method instead of `site::visit()`

Reply:
It will not work for cached page when we trigger it in `site::visit()`.

Reply:
Depending on the cache type (e.g. staticache), this won't work either way. So maybe it's better not expect this hook to work for cached pages.

Reply:
Understood. What if we put it on the routes? Do you think it makes sense?

Reply:
I agree with Nico that the hook shouldn't be relied on for cached pages. After all the purpose of the cache is to speed up processing as much as possible. Also, they still won't be called if Staticache is used (no matter if we put them in the routes or keep them here).

Which use case do you see for the `page:visit` hook?

Reply:
> Which use case do you see for the page.visit hook?

For example; For updating or processing a data via the plugin when the page is visited.

Reply:
Can that work at all when the cache is active?

Reply:
No, but it will work if we put it in routes. Also it will not work when there is a static cache. But when there is static cache, nothing works anyway. Controllers, models, routes, etc. So it's not just `page.visit`.

Reply:
That's right. 👍

Reply:
So where do you think the `page.visit` hook should be triggered? What do you think about having it in `App::resolve()`?